Libraries/LLM/Evaluate.swift

```diff
@@ -178,24 +178,16 @@ public enum GenerateDisposition {
 public func generate(
     promptTokens: [Int], parameters: GenerateParameters, model: LLMModel, tokenizer: Tokenizer,
     extraEOSTokens: Set<String>? = nil,
-    didGenerate: ([Int]) async -> GenerateDisposition
-) async -> GenerateResult {
+    didGenerate: ([Int]) -> GenerateDisposition
+) -> GenerateResult {
     var start = Date.timeIntervalSinceReferenceDate
     var promptTime: TimeInterval = 0
 
     // build a set of additional stop tokens
     let additionalEOSTokenIds = Set(
         (extraEOSTokens ?? [])
-            .map {
-                tokenizer.encode(text: $0)
-            }
-            .filter {
-                // discard anything that is not a single token.  sometimes
-                // the tokenizer will insert a <s> token, so accept that too
-                $0.count == 1 || ($0.count == 2 && $0[0] == 1)
-            }
-            .map {
-                $0.last!
+            .compactMap {
+                tokenizer.convertTokenToId($0)
```

Comment:
This is actually unrelated -- I was messing around with https://github.com/ml-explore/mlx-swift-examples/issues/89

Probably need to revert this and take it in a separate PR